hardware/simulation/icarus/Makefile

```diff
@@ -15,17 +13,27 @@ TB ?= $(CACHE_TB_DIR)/pipeline-iob-cache_tb.v
 #icarus verilog simulator
 VLOG = 	iverilog -W all -g2005-sv
 
```

Comment:
I propose to replace all lines below with just the following

run:
        $(VLOG) $(INCLUDE) $(DEFINE) $(VSRC)
        ./a.out
ifeq ($(VCD),1)
	if [ "`pgrep -u $(USER) gtkwave`" ]; then killall -q -9 gtkwave; fi
	gtkwave -a ../waves.gtkw uut.vcd &
endif

clean: gen-clean
	*.vcd a.out

.PHONY: run clean

Reply:
sorry, github missed the indentation

gen-clean should exist in config.mk of all repos since many makefiles use it:
gen-clean:
	@rm -f *# *~

Reply:
also I suggest moving .PHONY: run clean to before the phony targets

Reply:
run target: in this way, vcd will not be generated unless VCD var is set to 1 by default. But this not always what a user will be interested in.

clean: I suggest to replace *.vcd by uut.vcd

.PHONY: OK

Reply:
In all other repositories if you want vcd you need to make VCD=1, for example 

make sim VCD=1


in the command line directly


ok for uut.vcd

Reply:
Ok for make sim VCD=1
It works well with the suggested solution